### PR TITLE
bindings rmi test: clarify behavior

### PR DESCRIPTION
**Pull request by vrothberg**
The libpod/images/remove endpoint is not very REST-ish but, after some
debate, was decided for it to be implemented as is for the following reasons.

First, it allows for batch removing images which improves performance
significantly.  Note that Docker does support `rmi -a`!

Second, it allows for hiding the logic of setting the right exit code to
use from the client and keep all the logic on the server.

Hence, when removing an image that does not exist, the server will
return a 200. The response, however, includes the error message to be
used *and* the exit code that podman-remote will use.

Fixes: #12441
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

@edsantiago PTAL

### CODE REVIEW

**Comment by vrothberg** [comment]
Swagger docs: https://github.com/containers/podman/blob/main/pkg/api/server/register_images.go#L944

File: pkg/bindings/test/images_test.go [hunk 1]
```diff
@@ -85,14 +85,16 @@ var _ = Describe("Podman images", func() {
...
+		// images for performance reasons and for hiding the logic of
+		// deciding which exit code to use from the client.
 		response, errs := images.Remove(bt.conn, []string{"foobar5000"}, nil)
 		Expect(len(errs)).To(BeNumerically(">", 0))
-		code, _ := bindings.CheckResponseCode(errs[0])
-		// FIXME FIXME FIXME: #12441: THIS IS BROKEN
-		// FIXME FIXME FIXME: we get msg: "foobar5000: image not known"
-		// FIXME FIXME FIXME: ...with no ResponseCode
-		Expect(code).To(BeNumerically("==", -1))
+		Expect(response.ExitCode).To(BeNumerically("==", -1)) // podman-remote would exit with -1
```
Comment (Luap99) [inline pkg/bindings/test/images_test.go:97]: Why is the exit code -1? Exit codes should be between 0-255 AFAIK

Comment (vrothberg) [reply pkg/bindings/test/images_test.go:97]: It should be `1` in this case. Tests are also failing.

**Comment by TomSweeneyRedHat** [comment]
Changes look OK to me, but the tests aren't at all happy.

**Comment by vrothberg** [comment]
> Changes look OK to me, but the tests aren't at all happy.

search on registry.access.redhat.com/* is broken. All tests in all PRs are red at the moment :^)

**Comment by rhatdan** [comment]
LGTM

**Comment by vrothberg** [comment]
rebased to fix the search flake :heavy_check_mark:

**Comment by edsantiago** [comment]
Unintended consequences
```
         # # podman rmi quay.io/libpod/testimage@sha256:d48f2feaca74863c342cd9ce11edbe208675975740e7f4dd635b7b345339426a
         # Untagged: quay.io/libpod/testimage@sha256:d48f2feaca74863c342cd9ce11edbe208675975740e7f4dd635b7b345339426a
         # time="2021-12-16T06:55:11-06:00" level=error msg=0
         # #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
         # #|     FAIL: podman rmi untags, does not remove
         # #| expected: 'Untagged: quay.io/libpod/testimage@sha256:d48f2feaca74863c342cd9ce11edbe208675975740e7f4dd635b7b345339426a'
         # #|   actual: 'Untagged: quay.io/libpod/testimage@sha256:d48f2feaca74863c342cd9ce11edbe208675975740e7f4dd635b7b345339426a'
         # #|         > 'time="2021-12-16T06:55:11-06:00" level=error msg=0'
         # #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: cmd/podman/images/rm.go [hunk 2]
```diff
@@ -80,6 +81,7 @@ func rm(cmd *cobra.Command, args []string) error {
 				fmt.Println("Deleted: " + d)
 			}
 		}
+		logrus.Errorf("%d", report.ExitCode)
```
Comment (vrothberg) [inline cmd/podman/images/rm.go:84]: D'oh!

**Comment by vrothberg** [comment]
ready to merge

**Pull request merged**